### PR TITLE
[WEB-367] Add internal links on the website Socket.io concept

### PR DESCRIPTION
**Pull request by Nosfheratu**
Explain why this change is being made.
----
Links to Socket.io concept page was not present in the content of the website

Explain how this is accomplished.
----
Added Socket.io links to content of the textile files

How do you manually test this?
----
Navigate to `/concepts/feathersjs`
Navigate to `/concepts/websockets`

Look for `Socket.io` links should navigate to `/concepts/socketio`

Merge/Deploy Checklist
----

- [ ] It has been visually reviewed and accepted.
- [ ] It has been reviewed for code quality.
- [ ] It adds tests for the implemented features/fixed bugs.

Frontend Checklist
----

- [ ] Does this work on major browsers (Chrome, Firefox, Safari)?
- [ ] Does this work on mobile?
- [ ] Have you compared with the initial design / fit our [brand guidelines](https://brand.ably.com/)?
- [ ] Have you checked the code for accessibility issues?

### CODE REVIEW

File: content/concepts/feathersjs.textile [hunk 1]
```diff
@@ -268,4 +268,4 @@ The problem with adding load balancers and multiple servers is the associated co
 
 "Ably’s engineering blog":https://www.ably.io/blog/tag/ably-engineering/ describes problems and solutions associated with scaling realtime apps. When you find your (or your team’s) time taken up solving and/ or provisioning for these issues, that’s when - we believe - it’s worth looking for a cloud or better still serverless solution, leaving you free to concentrate on the core functionality of your app. That said, for small-scale applications or developers starting out with realtime - until realtime functionality scales and gets complicated, there is indeed a point at which FeathersJS could fit the bill of the holy grail of frameworks. Developers can judge for themselves when holy grail status starts to fade, and think about how to move on.
 
-*"Ably":https://www.ably.io/ is a global cloud network for streaming data and managing the full lifecycle of realtime APIs. To talk about scaling your app or service, get in touch with our dev-rel team.*
\ No newline at end of file
+*"Ably":https://www.ably.io/ is a global cloud network for streaming data and managing the full lifecycle of realtime APIs. To talk about scaling your app or service, get in touch with our dev-rel team.*
```
Comment (remotekieran) [inline content/concepts/feathersjs.textile:271]: I realise this is not part of the original ticket, but I see this is out of date and that we're asking people to contact the Dev Rel team but provide no contact link. Could you replace this paragraph with:

"Ably":https://ably.com is a realtime messaging platform. We deliver billions of realtime messages every day to more than 50 million end-users across web, mobile, and IoT platforms.

Developers use Ably’s 40+ SDKs to build realtime capabilities in their apps with our "multi-protocol pub/sub messaging":https://ably.com/channels, "presence":https://www.ably.io/documentation/core-features/presence, "push notifications":https://ably.com/push-notifications, "free streaming data sources":https://ably.com/hub from across industries like transport and finance, and "integrations that extend Ably":https://ably.com/reactor into third-party clouds and systems like AWS Kinesis and RabbitMQ.

To talk about scaling your realtime capabilities, get in touch with our "Developer Relations team":https://ably.com/contact.

File: content/concepts/websockets.textile [hunk 2]
```diff
@@ -427,7 +427,7 @@ h3(#socketio). Socket.io
 
 "Socket.io":https://github.com/socketio/socket.io has been around for a while now and could be thought of as the "jQuery" of WebSockets. It uses "long polling":/concepts/long-polling and WebSockets for its transports, by default starting with "long polling":/concepts/long-polling and then upgrading to WebSockets if available. Given the waning relevance of "long polling":/concepts/long-polling, Socket.io's main drawcards these days are its "other features":https://stackoverflow.com/questions/38546496/moving-from-socket-io-to-raw-websockets/38546537#38546537, such as restoring dropped connections, automatic support for JSON, and "namespaces", which are essentially isolated messaging channels multiplexed over the same client connection.
```
Comment (remotekieran) [inline content/concepts/websockets.textile:428]: Can you please replace this "Socket.io":https://github.com/socketio/socket.io with "Socket.io":https://ably.com/concepts/socketio

File: content/concepts/websockets.textile [hunk 3]
```diff
@@ -427,7 +427,7 @@ h3(#socketio). Socket.io
...
 
-Socket.io isn't actually interchangeable with generic WebSockets solutions - either on the server side or on the client side - and attempting to connect to something other than a Socket.io client or server will fail. It has its own additional handshake protocol, and some additional metadata included in each message.
+"Socket.io":https://www.ably.io/concepts/socketio isn't actually interchangeable with generic WebSockets solutions - either on the server side or on the client side - and attempting to connect to something other than a Socket.io client or server will fail. It has its own additional handshake protocol, and some additional metadata included in each message.
```
Comment (remotekieran) [inline content/concepts/websockets.textile:430]: Can you please link the other socket.io mention in this paragraph

File: content/concepts/websockets.textile [hunk 4]
```diff
@@ -427,7 +427,7 @@ h3(#socketio). Socket.io
...
 
 Should you use it? On the plus side, in addition to "the simplicity of getting up and running":https://socket.io/get-started/chat/, it's well-established and there is a wealth of learning material out there if you get stuck. On the other hand, there have been plenty of reports of memory leaks and general performance issues over time, and despite there being "a mountain of issues":https://github.com/socketio/socket.io/issues in the Github repository, responses are few, and "commits and updates":https://github.com/socketio/socket.io/commits/master these days seem relatively infrequent. Just like jQuery, Socket.io is largely a product of a bygone era, and for new projects, you'd be better off with something a little more modern. jQuery does still have its fans though, so...
```
Comment (remotekieran) [inline content/concepts/websockets.textile:432]: Can you please replace the strike through links with the links in code wrap:

"the simplicity of getting up and running":~https://socket.io/get-started/chat/~ `https://ably.com/concepts/socketio`

"a mountain of issues":~https://github.com/socketio/socket.io/issues~ `https://www.ably.io/concepts/socketio#future`

"commits and updates":~https://github.com/socketio/socket.io/commits/master~ `https://www.ably.io/concepts/socketio#future`

File: content/concepts/feathersjs.textile (hunk 1 above)
Comment (Nosfheratu) [reply content/concepts/feathersjs.textile:271]: Sure, no problem

**Comment by Nosfheratu** [comment]
@remotekieran added requested changes

File: content/concepts/feathersjs.textile [hunk 5]
```diff
@@ -29,7 +29,7 @@ What makes FeathersJS different from all the other JavaScript frameworks out the
...
+At the heart of FeathersJS are services. Services are simply JavaScript objects or ES6 classes that represent an API endpoint. Each one implements a specific set of functions to perform CRUD operations on the data. To add on to services, Feathers offers hooks which act as middleware. These can be made to execute code before or after a service function is called, or when an error occurs within a service function. FeathersJS also sends out events, if set up, when CRUD operations are executed by services. When used with "Socket.IO":https://www.ably.io/concepts/socketio, WebSockets can be used to emit and receive these events allowing FeathersJS to communicate between client and server creating realtime functionality. Again, every part of Feathers is made with the goal of being easy to understand, maintain, apply - and above all adapt and add on to.
```
Comment (tomczoink) [inline content/concepts/feathersjs.textile:32]: It'd be good to make local references to things within docs, so `https://www.ably.io/concepts/socketio` => `concepts/socketio`

File: content/concepts/feathersjs.textile [hunk 6]
```diff
@@ -268,4 +268,8 @@ The problem with adding load balancers and multiple servers is the associated co
 
 "Ably’s engineering blog":https://www.ably.io/blog/tag/ably-engineering/ describes problems and solutions associated with scaling realtime apps. When you find your (or your team’s) time taken up solving and/ or provisioning for these issues, that’s when - we believe - it’s worth looking for a cloud or better still serverless solution, leaving you free to concentrate on the core functionality of your app. That said, for small-scale applications or developers starting out with realtime - until realtime functionality scales and gets complicated, there is indeed a point at which FeathersJS could fit the bill of the holy grail of frameworks. Developers can judge for themselves when holy grail status starts to fade, and think about how to move on.
 
-*"Ably":https://www.ably.io/ is a global cloud network for streaming data and managing the full lifecycle of realtime APIs. To talk about scaling your app or service, get in touch with our dev-rel team.*
\ No newline at end of file
+"Ably":https://ably.com is a realtime messaging platform. We deliver billions of realtime messages every day to more than 50 million end-users across web, mobile, and IoT platforms.
```
Comment (tomczoink) [inline content/concepts/feathersjs.textile:271]: @kennethkalmer are we wanting these to link to ably.com already, or keep to ably.io until we properly transition?

Comment (kennethkalmer) [reply content/concepts/feathersjs.textile:271]: @tomczoink @Nosfheratu lets keep it to `.io` for now.

File: content/concepts/feathersjs.textile (hunk 5 above)
Comment (Nosfheratu) [reply content/concepts/feathersjs.textile]: fixed

File: content/concepts/feathersjs.textile (hunk 6 above)
Comment (MarkWoulfeAbly) [reply content/concepts/feathersjs.textile:271]: @kennethkalmer should @Nosfheratu go through and change these to .io now so we can land this PR?

**Comment by Nosfheratu** [comment]
@MarkWoulfeAbly the branch was rebased and also domain extension was fixed

File: content/concepts/feathersjs.textile (hunk 5 above)
Comment (MarkWoulfeAbly) [inline content/concepts/feathersjs.textile]:
```suggestion
At the heart of FeathersJS are services. Services are simply JavaScript objects or ES6 classes that represent an API endpoint. Each one implements a specific set of functions to perform CRUD operations on the data. To add on to services, Feathers offers hooks which act as middleware. These can be made to execute code before or after a service function is called, or when an error occurs within a service function. FeathersJS also sends out events, if set up, when CRUD operations are executed by services. When used with "Socket.IO":/concepts/socketio, WebSockets can be used to emit and receive these events allowing FeathersJS to communicate between client and server creating realtime functionality. Again, every part of Feathers is made with the goal of being easy to understand, maintain, apply - and above all adapt and add on to.
```

File: content/concepts/websockets.textile (hunk 3 above)
Comment (MarkWoulfeAbly) [inline content/concepts/websockets.textile]:
```suggestion
"Socket.io":/concepts/socketio isn't actually interchangeable with generic WebSockets solutions - either on the server side or on the client side - and attempting to connect to something other than a Socket.io client or server will fail. It has its own additional handshake protocol, and some additional metadata included in each message.
```

**Comment by MarkWoulfeAbly** [comment]
Thanks @Nosfheratu, I've just noticed you're still pushing commits, please give me a heads up when this is ready for final review.

**Comment by Nosfheratu** [comment]
> Thanks @Nosfheratu, I've just noticed you're still pushing commits, please give me a heads up when this is ready for final review.

Hey @MarkWoulfeAbly I think is done now...

File: content/concepts/feathersjs.textile [hunk 7]
```diff
@@ -264,8 +264,12 @@ h2(#feathersjs-cost). Freedom - but at what cost?
...
 
-*"Ably":https://www.ably.io/ is a global cloud network for streaming data and managing the full lifecycle of realtime APIs. To talk about scaling your app or service, get in touch with our dev-rel team.*
+"Ably":https://ably.io is a realtime messaging platform. We deliver billions of realtime messages every day to more than 50 million end-users across web, mobile, and IoT platforms.
+
+Developers use Ably’s 40+ SDKs to build realtime capabilities in their apps with our "multi-protocol pub/sub messaging":/channels, "presence":/documentation/core-features/presence, "push notifications":/push-notifications, "free streaming data sources":/hub from across industries like transport and finance, and "integrations that extend Ably":/reactor into third-party clouds and systems like AWS Kinesis and RabbitMQ.
```
Comment (MarkWoulfeAbly) [inline content/concepts/feathersjs.textile:273]:
```suggestion
Developers use Ably’s 40+ SDKs to build realtime capabilities in their apps with our "multi-protocol pub/sub messaging":/channels, "presence":/core-features/presence, "push notifications":/push-notifications, "free streaming data sources":/hub from across industries like transport and finance, and "integrations that extend Ably":/reactor into third-party clouds and systems like AWS Kinesis and RabbitMQ.
```

Comment (Nosfheratu) [reply content/concepts/feathersjs.textile:273]: @MarkWoulfeAbly https://www.ably.io/core-features/presence will show me 404 page

Comment (Nosfheratu) [reply content/concepts/feathersjs.textile:273]: https://www.ably.io//documentation/core-features/presence will display the correct page, so your suggested code does not apply

Comment (MarkWoulfeAbly) [reply content/concepts/feathersjs.textile:273]: @Nosfheratu the link is actually broken as it is currently if you go to the page in the deployment and try and click on it. Every URL in the documentation repo will automatically generate the necessary /documentation part so you just need to use the part after, in this case, /core-features/presence.

File: content/concepts/websockets.textile [hunk 8]
```diff
@@ -568,11 +568,11 @@ Such concerns are deserving of an article all of their own, and you'll find plen
...
+Ably is a battle-tested, enterprise-ready, cloud-based messaging platform with native support for WebSockets, "long polling":/concepts/long-polling?utm_source=websockets&utm_medium=concepts and other fallback transports, in addition to support for other "open protocols such as MQTT":/adapters?utm_source=websockets&utm_medium=concepts. We've already done all of the above work for you, which means you can get on with the real task of building your business's application platform and leave the mammoth task of implementing robust, reliable realtime messaging to us.
 
-Find out more about the "Ably Realtime platform":https://www.ably.io/dsn?utm_source=websockets&utm_medium=concepts.
+Find out more about the "Ably Realtime platform":/dsn?utm_source=websockets&utm_medium=concepts.
```
Comment (MarkWoulfeAbly) [inline content/concepts/websockets.textile:575]: This actually needs to be the full URL because the page does not exist within /documentation

Comment (Nosfheratu) [reply content/concepts/websockets.textile:575]: gotcha, changing now...

Comment (MarkWoulfeAbly) [reply content/concepts/websockets.textile:575]:
```suggestion
Find out more about the "Ably Realtime platform":https://www.ably.io/dsn?utm_source=websockets&utm_medium=concepts.
```

Comment (MarkWoulfeAbly) [reply content/concepts/websockets.textile:575]: @Nosfheratu you can accept my suggestion above and then we are good to go.

**Comment by MarkWoulfeAbly** [comment]
Thanks @Nosfheratu, apologies for the confusing URL rules, will hopefully update the documentation on this.

**Pull request merged**